internal/helpers/secrets.go

```diff
@@ -38,21 +37,46 @@ var OwnerLabels = map[string]string{
 	"app.kubernetes.io/component":  "secret-sync",
 }
 
-func getOwnerRefFromObj(owner ctrlclient.Object, scheme *runtime.Scheme) (metav1.OwnerReference, error) {
-	ownerRef := metav1.OwnerReference{
-		Name: owner.GetName(),
-		UID:  owner.GetUID(),
-	}
+// SyncableSecretMetaData provides common data structure that extracts the bits pertinent
+// when handling any of the sync-able secret custom resource types.
+//
+// See NewSyncableSecretMetaData for the supported object types.
+type SyncableSecretMetaData struct {
+	// APIVersion of the syncable-secret object. Maps to obj.APIVersion.
+	APIVersion string
+	// Kind of the syncable-secret object. Maps to obj.Kind.
+	Kind string
+	// Destination of the syncable-secret object. Maps to obj.Spec.Destination.
+	Destination *secretsv1beta1.Destination
+}
 
-	gvk, err := apiutil.GVKForObject(owner, scheme)
-	if err != nil {
-		return ownerRef, err
+// NewSyncableSecretMetaData returns SyncableSecretMetaData if obj is a supported type.
+// An error will be returned of obj is not a supported type.
+//
+// Supported types for obj are: VaultDynamicSecret, VaultStaticSecret. VaultPKISecret
+func NewSyncableSecretMetaData(obj ctrlclient.Object) (*SyncableSecretMetaData, error) {
```

Comment:
I think this is a merge issue, we recently moved `NewSyncableSecretMetadata` to `internal/common/common.go` and modified it to return `AuthRef` as well, see here: https://github.com/hashicorp/vault-secrets-operator/blob/8a710f0feb6e4e1bd6f1aba816de43a1cd29f8eb/internal/common/common.go#L322

This change will probably break namespaces?